### ISSUE TITLE
Improve devapp build and load time

### ISSUE THREAD

**Issue by jelbourn** [angular/components#13451]
Some possible ways to do this:
- [x] Enable AoT
- [x] Lazy-load demos on each page
- [ ] Only rerun `ngc` gulp tasks for a given directory if any files there have changed
- [x] Full up switch devapp to running via bazel
- [ ] Full up switch to webpack

Work done to move Bazel dev-app forward:

* https://github.com/angular/angular/commit/0ada23a5fb9c2e0ebd33301d192febefa309ddb9
* https://github.com/angular/angular/commit/d3c08e74f626c75a262535d4e78c251f97855d02
* etc.

Until switching to Bazel which is blocked on making the devserver work on Windows:

- Use UMD bundle
- Inline resources

**Comment by devversion** [comment]
This is done now. I think we are in pretty good-shape in terms of lazy loading. Building / rebuilding is as fast as it gets based on `@angular/bazel/ngc-wrapped`. We also fixed an incrementality bug recently that caused unnecessary rebuilds w/ ibazel.

We could improve things slightly by rethinking how our components-examples package is built up, but that seems out-of-scope for this issue and rather related to how we'll handle docs in the future.
